### ISSUE TITLE
Regarding keeping editor and tooling info on the front-page up-to-date

### ISSUE THREAD

**Issue by sogaiu** [janet-lang/janet-lang.org#222]
I've noticed over time that certain kinds of editor-related questions arise in the various channels of communication.  A characteristic of some of these is that they have to do with older (sometimes not-maintained-so-actively) tooling.

My suspicion is that some of the tooling tried is based on seeing [this portion](https://janet-lang.org/#Modules-and-Libraries) of the top page of janet-lang.org (the part under "For editor support").  I think that list is (and has been) missing some things [1] but it also has some bits that may be aren't what you'd want to try first.

Updating the list with specifics is one idea, but this site doesn't necessarily get updated that frequently and inevitably the tooling situation is likely to change.

I think it would be nicer if people understood that the tooling situation is an evolving one and that more up-to-date info might be obtained via the Zulip instance (or perhaps the discussion area of the janet repository).

I think it's unfortunate when someone spends time on a tool only to realize later that if they had known of other options they would have made a different decision (at least with respect to what to try and in what order).  This situation is particularly unfortunate when the initial choices don't work out so well.

Specifically, I think it might make sense to have a link to [the "editors and tooling" channel on our Zulip instance](https://janet.zulipchat.com/#narrow/stream/409409-editors-and-tooling) and/or one suggesting the janet repository discussion area along with some brief explanation.

This idea has the benefit of not necessarily requiring much updating (at least compared with looking after a specific list) and it points people at locations that might provide further info. 

Does this sound like a reasonable idea?

---

[1] Some things that are missing:

* CFiggers' vscode-janet-plus-plus
* Helix has some Janet support
* Some additional Emacs bits
* A couple of trial kits (Emacs and Neovim)

There are also other things that may be relevant related to LSP, linting, diffing, etc. that are not listed.

**Comment by bakpakin** [comment]
There is definitely some work to do here, and really I need to go an do some maintenance on the website to update that info. I personally maintain and use only some of the tooling, so it needs to be up to other people to let me know of tooling that they use and its quality.

As for the trial kits, we should update that.

**Comment by sogaiu** [comment]
Thanks for the feedback.

I'll work on gathering a list of bits that might be relevant.  Perhaps it can be used as a source for one or more updates.

**Comment by sogaiu** [comment]
I've made an initial summary [here](https://github.com/sogaiu/janet-editor-and-tooling-info).

I'll ask around a bit to see if other useful things turn up.

**Comment by sogaiu** [comment]
I gave the current situation a bit more thought.

If only listing editors, there are some that it would be easy to just have a link for, e.g.

* CudaText (link to [issue in janet's repository](https://github.com/janet-lang/janet/issues/1107))
* Helix (link to [helix's repo](https://github.com/helix-editor/helix/) or home page)
* TIC-80 (contains an editor; just link to [TIC-80's repo](https://github.com/nesbox/TIC-80/))
* Sublime Text (link to [sublime-janet](https://github.com/archydragon/sublime-janet))
* Vim (link to [janet.vim](https://github.com/janet-lang/janet.vim))

Not so clear are the remaining:

* Emacs
* Kakoune
* Neovim
* VSCode

Kakoune [1] and VSCode each have a couple of efforts, so the single link approach may not be appropriate.  I suppose since it's just two links each, may be that's ok.

However, Emacs and Neovim have a variety of pieces that probably make sense for people to be aware of -- I suppose if the trial kits were linked to, I could try to provide info about the various components of interest at the trial kit repositories...

---

[1] I'm not sure if both Kakoune efforts are currently recommended.  I'm waiting on a response about this.

**Comment by pepe** [comment]
I am on pause with Kakoune. But I think it is still relevant.

**Comment by sogaiu** [comment]
@pepe Thanks for the follow-up -- so do you think both  [JaKoune](https://git.sr.ht/~pepe/jakoune) and [janet.kak](https://github.com/pepe/janet.kak) should be listed?

**Comment by pepe** [comment]
To be honest, I'm not sure. I will make sure this week, and let you know

**Comment by sogaiu** [comment]
> However, Emacs and Neovim have a variety of pieces that probably make sense for people to be aware of -- I suppose if the trial kits were linked to, I could try to provide info about the various components of interest at the trial kit repositories...

I've adapted this idea a bit so that in the top section of the janet-neovim-trial-kit README, there is this:

> Or...learn about existing Neovim plugins with Janet features by reading some [docs](https://github.com/sogaiu/janet-editor-and-tooling-info/blob/master/doc/neovim.md).

The link should take one to a page that describes some Janet-related plugins for Neovim.  Perhaps this is a good compromise.

I plan to do something similar for janet-emacs-trial-kit.

**Comment by sogaiu** [comment]
I've expanded the purposes of the two trial kits to also provide some guidance [1] about what plugins / packages exist for the two respective editors.  If the trial kits are linked to from janet-lang.org, then followers of such links should end with some relevant hints. 

A bit indirect perhaps...

---

One idea about what to do for Kakoune and VSCode is to link to [this page](https://github.com/sogaiu/janet-editor-and-tooling-info/blob/master/doc/kakoune.md) and [this page](https://github.com/sogaiu/janet-editor-and-tooling-info/blob/master/doc/vscode.md), repsectively.

If that's palatable, possibly a similar thing could be done for Emacs and Neovim...since they both have pages ([Emacs](https://github.com/sogaiu/janet-editor-and-tooling-info/blob/master/doc/emacs.md), [Neovim](https://github.com/sogaiu/janet-editor-and-tooling-info/blob/master/doc/neovim.md)) similar to those for Kakoune and VSCode.

---

[1] Actually, I cheated by linking to certain parts of [janet-editor-and-tooling-info](https://github.com/sogaiu/janet-editor-and-tooling-info/) :P

**Comment by pepe** [comment]
After some thinking, I would say that neither janet.kak nor JaKoune should be part of the documentation. First is superseded by the kakoune default janet.kak, and second, I will sunset soon.

**Comment by sogaiu** [comment]
Thanks for looking into the situation.

So to summarize then, in Kakoune's case, we can just say that there is functionality built into the editor itself?

**Comment by pepe** [comment]
Your findings are correct. It is filetype detection, syntax highlighting and indentation.

**Comment by sogaiu** [comment]
Thanks a lot for the clarification.

**Comment by sogaiu** [comment]
Ok, may be I'll work on a draft PR that tries to incorporate some of the above.

**Comment by sogaiu** [comment]
It may be obvious, but FWIW, #225 is a draft PR for this issue.

**Comment by sogaiu** [comment]
Closing, as #225 has been merged.
